tests/lax_scipy_sparse_test.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import partial
+from absl.testing import parameterized
+from absl.testing import absltest
+from scipy.sparse import spdiags
+import numpy as onp
+import scipy.sparse.linalg as osp_sparse
+from jax import lax
+from jax import test_util as jtu
+from jax.experimental import sparse
+
+float_types = [onp.float32, onp.float64]
+complex_types = [onp.complex64, onp.complex128]
+
+class LaxBackedScipyTests(jtu.JaxTestCase):
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_{}D_{}".format(dim, onp.dtype(dtype).name),
+    "dim": dim, "dtype": dtype}
+    for dim in [3, 7, 100, int(1e4)]
+    for dtype in float_types + complex_types))
+  def test_linalg_sparse_cg(self, dim, dtype):
+    def high_precision_dot(a, b):
+      return lax.dot(a, b, precision=lax.Precision.HIGHEST)
+
+    def build_and_solve(a, b):
+      # intentionally non-linear in a and b
+      matvec = partial(high_precision_dot, a)
+      return sparse.cg(matvec, b)
+
+    r = onp.random.RandomState(dim)
```

Comment:
Take a look at the `rng_factory` argument from other JAX tests, e.g., in `linalg_test.py`.

Reply:
thanks! how do i "zero" out values in my dense matrix to create a sparse one in JAX ?

Reply:
`rng(shape, dtype)` returns a normal NumPy array. So you could modify it in-place if you like, but I don't think it's essential to test this function with actual "sparse" arrays. We don't really have sparse arrays in JAX currently, functions that compute a matrix-vector product are probably the closest thing.